X9.146/README.md

```diff
@@ -0,0 +1,406 @@
+
+## Post-Quantum
+
+Tested with these wolfSSL build options:
+
+```sh
+./autogen.sh  # If cloned from GitHub
+./configure --enable-x9-146 --with-liboqs --enable-debug
+make
+sudo make install
+sudo ldconfig # required on some targets
+```
+
+In the directory where this README.md file is found, clean up previous build
+products and certificates and then build the applications.
+
+```sh
+make clean all
+```
+NOTE: `clean` removes certificates and keys in this directory.
+
+### What to Expect
+
+There will be a lot of debug output going to stderr. On the client side, during
+the call to `DoTls13Certificate()`, please search for the following messages to
+confirm that the alternative signature was verified:
+
+```
+Alternative signature has been verified!
+Verified Peer's cert
+```
+
+These debug messages indicate that the client has verified the alternative
+post-quantum certificate chain. The second message indicates that normal
+verification was also successful.
+
+On the client side, during the call to `DoTls13CertificateVerify()` look for
+messages that indicate both conventional and post-quantum verification:
+
+For example, if you are doing ECDSA with Falcon, you will see the following:
+
+```
+Doing ECC peer cert verify
+wolfSSL Entering EccVerify
+wolfSSL Leaving EccVerify, return 0
+Doing Falcon peer cert verify
+wolfSSL Leaving DoTls13CertificateVerify, return 0
+```
+
+### ECDSA Demos
+
+#### P-256 and Dilithium Level 2 Demo
+Generate the various conventional keys; the post-quantum key are pre-generated:
+
+```sh
+openssl genpkey -algorithm ec -pkeyopt ec_paramgen_curve:P-256 -out ca-key.der -outform der
+
+openssl genpkey -algorithm ec -pkeyopt ec_paramgen_curve:P-256 -out server-key.der -outform der
+```
+
+Generate the certificate chain:
+
+```
+./gen_ecdsa_dilithium_dual_keysig_root_cert 2
+
+./gen_ecdsa_dilithium_dual_keysig_server_cert 2
+```
+
+Convert the DER encoded resulting certificates and keys into PEM:
+
+```
+openssl x509  -in ca-cert-pq.der -inform der -out ca-P256-dilithium2-cert.pem -outform pem
+
+openssl x509  -in server-cert-pq.der -inform der -out server-P256-dilithium2-cert.pem -outform pem
+
+openssl pkey  -in server-key.der -inform der -out server-P256-key.pem -outform pem
+
+openssl pkey  -in ../certs/dilithium_level2_server_key.der -inform der -out server-dilithium2-key-pq.pem -outform pem
+i
+(last one must be done with OQS's openssl fork)
+```
+Then in wolfssl's source directory:
+
+```
+examples/server/server -v 4 -c ../wolfssl-examples/X9.146/server-P256-dilithium2-cert.pem -k ../wolfssl-examples/X9.146/server-P256-key.pem --altPrivKey ../wolfssl-examples/X9.146/server-dilithium2-key-pq.pem
+
+examples/client/client -v 4 -A ../wolfssl-examples/X9.146/ca-P256-dilithium2-cert.pem
+```
+
+#### P-384 and Dilithium Level 3 Demo
+Generate the various conventional keys; the post-quantum key are pre-generated:
+
+```sh
+openssl genpkey -algorithm ec -pkeyopt ec_paramgen_curve:P-384 -out ca-key.der -outform der

```

Comment:
Lets reference wolfCLU where it can be used.

Reply:
Hesitant to require users to build wolfclu.  It would be different if they could do it all with wolfclu but they will still need openssl.